src/v/model/tests/token_bucket_test.cc

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "model/token_bucket.h"
+
+#include <seastar/core/sleep.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+#include <chrono>
+using namespace std::chrono_literals;
+
+constexpr size_t RATE = 25;
+
+SEASTAR_THREAD_TEST_CASE(test_simple_throttle) {
+    auto throttler_instance = throttler(RATE, "test_simple_throttle");
```

Comment:
Can we make the `token_bucket` testable by providing a way to inject `clock` then we could use `ss::manual_clock` to make the test deterministic.

Reply:
fixed